bindings/src/test/java/io/pravega/storage/filesystem/FileSystemLockTest.java

```diff
@@ -0,0 +1,139 @@
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Unit tests for FileSystemBasedLock.
+ */
+public class FileSystemLockTest {
+
+    static final long ONE_HOUR = Duration.of(1, ChronoUnit.HOURS).toMillis();
+    static final long ONE_MILLI = Duration.of(1, ChronoUnit.MILLIS).toMillis();
+    static final long TWO_MILLI = Duration.of(2, ChronoUnit.MILLIS).toMillis();
+    static final long HUNDRED_MILLI = Duration.of(100, ChronoUnit.MILLIS).toMillis();
+
+    private File baseDir = null;
+    private AtomicBoolean inCriticalSection = new AtomicBoolean();
+    private Random rnd = new Random();
+
+    @Before
+    public void setUp() throws Exception {
+        this.baseDir = Files.createTempDirectory("test_nfs").toFile().getAbsoluteFile();
+    }
+
+    @After
+    public void tearDown() {
+        FileHelpers.deleteFileOrDirectory(baseDir);
+        baseDir = null;
+    }
+
+    /**
+     * Test that lock file is created and deleted.
+     * @throws Exception
+     */
+    @Test()
+    public void testBasicLocking() throws Exception {
+        Path lockFilepath = Paths.get(baseDir.getAbsolutePath(), "test1");
+        FileSystemBasedLock lock = new FileSystemBasedLock(lockFilepath);
+        Assert.assertTrue("Lock file should be created after creating lock.", Files.exists(lockFilepath));
+        lock.close();
+        Assert.assertFalse("Lock file should be deleted after close.", Files.exists(lockFilepath));
+        // Calling close multiple times should be okay.
+        lock.close();
+    }
+
+    /**
+     * Create first lock but do not close it.
+     * Try to create second lock with same file, the lock should timeout.
+     * @throws Exception
+     */
+    @Test()
+    public void testConflict() throws Exception {
+        Path lockFilepath = Paths.get(baseDir.getAbsolutePath(), "test1");
+        FileSystemBasedLock lock1 = new FileSystemBasedLock(lockFilepath, ONE_HOUR, ONE_MILLI, TWO_MILLI);
+        Assert.assertTrue("Lock file should be created after creating lock.", Files.exists(lockFilepath));
+        try {
+            FileSystemBasedLock lock2 = new FileSystemBasedLock(lockFilepath, ONE_HOUR, ONE_MILLI, TWO_MILLI);
+            Assert.fail("Exception should be thrown");
+        } catch (TimeoutException ex) {
+            // good exception thrown as expected
+        } finally {
+            Files.delete(lockFilepath);
+        }
+    }
+
+    /**
+     * Create a test lock, but don't close it.
+     * Wait for it to expire.
+     * Create second lock, it should succeed.
+     * @throws Exception
+     */
+    @Test()
+    public void testExpiredLock() throws Exception {
+        Path lockFilepath = Paths.get(baseDir.getAbsolutePath(), "test1");
+        FileSystemBasedLock lock1 = new FileSystemBasedLock(lockFilepath, ONE_MILLI, ONE_MILLI, HUNDRED_MILLI);
+        Assert.assertTrue("Lock file should be created after creating lock.", Files.exists(lockFilepath));
+        Thread.sleep(5);
```

Comment:
An acquire API call would allow you to continuously try to acquire the lock and check that it eventually expires and is released. It is better than having a sleep in the test case.

Reply:
Trying to simulate expired lock by actually waiting for 5 ms.